src/pkg/util/env/clean.go

```diff
@@ -27,13 +28,35 @@ var alwaysPassKeys = map[string]bool{
 
 // SetContainerEnv cleans environment variables before running the container
 func SetContainerEnv(g *generate.Generator, env []string, cleanEnv bool, homeDest string) {
+
+	// first deal with special variables that allow user to control $PATH at
+	// runtime (meh... special cases)
+	if prependPath := os.Getenv("SINGULARITYENV_PREPEND_PATH"); prependPath != "" {
+		g.AddProcessEnv("SING_USER_DEFINED_PREPEND_PATH", prependPath)
+	}
+
+	if appendPath := os.Getenv("SINGULARITYENV_APPEND_PATH"); appendPath != "" {
+		g.AddProcessEnv("SING_USER_DEFINED_APPEND_PATH", appendPath)
+	}
+
+	if userPath := os.Getenv("SINGULARITYENV_PATH"); userPath != "" {
+		g.AddProcessEnv("SING_USER_DEFINED_PATH", userPath)
+	}
+
 	for _, env := range env {
 		e := strings.SplitN(env, "=", 2)
 		if len(e) != 2 {
 			sylog.Verbosef("Can't process environment variable %s", env)
 			continue
 		}
 
+		if e[0] == "SINGULARITYENV_PREPEND_PATH" ||
+			e[0] == "SINGULARITYENV_APPEND_PATH" ||
+			e[0] == "SINGULARITYENV_PATH" {
+			sylog.Verbosef("Not adding special case PATH control variable %s to container environment", e[0])
+			continue
+		}
```

Comment:
@ArangoGutierrez just moving the if statements above the loop did not fix the issue because `env` is passed as input to the function.  So a special case had to be added to the `for` loop.

Reply:
you could create const at the top for the 3 cases, and here just do 
```go
	if e[0] == prependPath || e[0] == apendPath || e[0] == envPath {
			sylog.Verbosef("Not adding special case PATH control variable %s to container environment", e[0])
			continue
		}
```

Reply:
I could.  Don't really know what it buys us though.  I feel like it is pretty readable the way it is.